client/components/data/domain-management/email/index.jsx

```diff
@@ -48,84 +51,125 @@ function getStateFromStores( props ) {
 	};
 }
 
-const EmailData = React.createClass( {
-	displayName: 'EmailData',
+export class EmailData extends Component {
+	static propTypes = {
```

Comment:
`component: React.PropTypes.func.isRequired` should still be a part of `propTypes`.